src/w3d/renderer/streakrender.h

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Streak Renderer
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "matrix3d.h"
+#include "shader.h"
+#include "sphere.h"
+#include "vector3.h"
+
+class TextureClass;
+struct VertexFormatXYZUV1;
+struct W3dEmitterLinePropertiesStruct;
+class RenderInfoClass;
+
+class StreakRendererClass
+{
+public:
+    StreakRendererClass();
+    StreakRendererClass(const StreakRendererClass &src);
+    StreakRendererClass &operator=(const StreakRendererClass &src);
+    ~StreakRendererClass();
+
+    enum TextureMapMode
+    {
+        UNIFORM_WIDTH_TEXTURE_MAP = 0x00000000,
+        UNIFORM_LENGTH_TEXTURE_MAP = 0x00000001,
+        TILED_TEXTURE_MAP = 0x00000002
+    };
+
+    void Init(const W3dEmitterLinePropertiesStruct &props);
+    void Set_Texture(TextureClass *texture);
+    const TextureClass *Get_Texture() const;
+    void Render(RenderInfoClass &rinfo,
+        const Matrix3D &transform,
+        unsigned int point_count,
+        Vector3 *points,
+        const SphereClass &obj_sphere);
+
+    void Subdivision_Util(unsigned int point_cnt,
+        const Vector3 *xformed_pts,
+        const float *base_tex_v,
+        unsigned int *p_sub_point_cnt,
+        Vector3 *xformed_subdiv_pts,
+        float *subdiv_tex_v);
+
+    void Render_Streak(RenderInfoClass &rinfo,
+        Matrix3D const &transform,
+        unsigned int point_count,
+        Vector3 *points,
+        Vector4 *colors,
+        float *widths,
+        SphereClass const &obj_sphere,
+        unsigned int *unk);
+
+    VertexFormatXYZUV1 *Get_Vertex_Buffer(unsigned int vertex_count);
+
+    TextureMapMode Get_Texture_Mapping_Mode() const;
+    void Set_Texture_Mapping_Mode(TextureMapMode mode);
+
+    int Is_Freeze_Random() const { return m_bits & FREEZE_RANDOM; }
+    int Is_Merge_Intersections() const { return m_bits & MERGE_INTERSECTIONS; }
+    int Is_Sorting_Disabled() const { return m_bits & DISABLE_SORTING; }
+    void Set_Color(const Vector3 &color) { m_color = color; }
+    void Set_Current_Subdivision_Level(unsigned int lv) { m_subdivisionLevel = lv; }

```

Comment:
probably move to cpp and make this the way it is in bfme2, with assert and setting to 0

Reply:
fixed.